kotlinx-coroutines-core/common/src/channels/Broadcast.kt

```diff
@@ -10,27 +10,49 @@ import kotlinx.coroutines.channels.Channel.Factory.UNLIMITED
 import kotlinx.coroutines.intrinsics.*
 import kotlin.coroutines.*
 import kotlin.coroutines.intrinsics.*
+import kotlin.native.concurrent.*
 
 /**
  * Broadcasts all elements of the channel.
+ * This function [consumes][ReceiveChannel.consume] all elements of the original [ReceiveChannel].
  *
  * The kind of the resulting channel depends on the specified [capacity] parameter:
  * when `capacity` is positive (1 by default), but less than [UNLIMITED] -- uses `ArrayBroadcastChannel` with a buffer of given capacity,
  * when `capacity` is [CONFLATED] -- uses [ConflatedBroadcastChannel] that conflates back-to-back sends;
  *   Note that resulting channel behaves like [ConflatedBroadcastChannel] but is not an instance of [ConflatedBroadcastChannel].
  *   otherwise -- throws [IllegalArgumentException].
  *
+ * ### Cancelling broadcast
+ *
+ * **To stop broadcasting from the underlying channel call [cancel][BroadcastChannel.cancel] on the result.**
+ *
+ * Do not use [close][BroadcastChannel.close] on the resulting channel.
+ * It causes eventual failure of the broadcast coroutine and cancellation of the underlying channel, too,
+ * but it is not as prompt.
+ *
+ * ### Future replacement
+ *
+ * This function has an inappropriate result type of [BroadcastChannel] which provides
+ * [send][BroadcastChannel.send] and [close][BroadcastChannel.close] operations that interfere with
+ * the broadcasting coroutine in hard-to-specify ways. It will be replaced with
+ * sharing operators on [Flow][kotlinx.coroutines.flow.Flow] in the future.
+ *
  * @param start coroutine start option. The default value is [CoroutineStart.LAZY].
  */
+//@ObsoleteCoroutinesApi // since version 1.4.0
```

Comment:
Why have you decided to leave `CoroutineScope.broadcastIn` obsolete, but not this one?

Reply:
I've changed my mind but forgot about the other version. Fixed. None of them will be obsolete.